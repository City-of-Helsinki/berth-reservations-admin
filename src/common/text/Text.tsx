import React from 'react';
import classNames from 'classnames';

import styles from './text.module.scss';

export interface TextProps {
  children: React.ReactNode;
  className?: string;
  color?: 'standard' | 'brand' | 'critical' | 'secondary' | 'info' | 'gray';
  as?: 'span' | 'em' | 'strong' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  size?: 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'xxl' | 'xxxl';
  weight?: 'bold' | 'normalWeight' | 'light';
  italic?: boolean;
  underlined?: boolean;
  uppercase?: boolean;
}

const Text = ({
  className,
  color = 'standard',
  size,
  as: Element = 'span',
  weight,
  italic,
  underlined,
  uppercase,
  children,
}: TextProps) => {
  return (
    <Element
      className={classNames(
        styles.text,
        styles[color],
        styles[Element],
        size && styles[size],
        weight && styles[weight],
        {
          [styles.italic]: italic,
          [styles.underlined]: underlined,
          [styles.uppercase]: uppercase,
        },
        className
      )}
    >
      {children}
    </Element>
  );
};

export default Text;
