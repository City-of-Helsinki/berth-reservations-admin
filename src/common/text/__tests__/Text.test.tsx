import React from 'react';
import { shallow } from 'enzyme';

import Text, { TextProps } from '../Text';

describe('Text', () => {
  const getWrapper = (props?: TextProps) => shallow(<Text {...props}>Test</Text>);

  it('renders normally', () => {
    const wrapper = getWrapper({ children: 'test' });
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render "h1" if the supplied "as" prop is "h1"', () => {
    const wrapper = getWrapper({ as: 'h1', children: 'test' });
    expect(wrapper.find('h1')).toHaveLength(1);
  });

  it('should add the supplied color prop as a className', () => {
    const color = 'standard';
    const wrapper = getWrapper({ children: 'test', color });
    expect(wrapper.hasClass(color)).toBe(true);
  });

  it('should add the supplied size prop as a className', () => {
    const size = 'xl';
    const wrapper = getWrapper({ children: 'test', size });
    expect(wrapper.hasClass(size)).toBe(true);
  });

  it('should add bold as a className if the supplied weight prop has a value of bold', () => {
    const wrapper = getWrapper({ weight: 'bold', children: 'test' });
    expect(wrapper.hasClass('bold')).toBe(true);
  });

  it('should add italic as a className if the supplied italic prop is true', () => {
    const wrapper = getWrapper({ children: 'test', italic: true });
    expect(wrapper.hasClass('italic')).toBe(true);
  });

  it('should add underlined as a className if the supplied underlined prop is true', () => {
    const wrapper = getWrapper({ children: 'test', underlined: true });
    expect(wrapper.hasClass('underlined')).toBe(true);
  });
});
