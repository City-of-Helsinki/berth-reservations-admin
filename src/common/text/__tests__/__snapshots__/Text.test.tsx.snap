// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders normally 1`] = `
<span
  class="text standard span"
>
  Test
</span>
`;
